backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/TransitionUsageImpl.java

```diff
@@ -270,15 +265,19 @@ public EList<AcceptActionUsage> getTriggerAction() {
     }
 
     /**
-     * <!-- begin-user-doc --> <!-- end-user-doc -->
+     * <!-- begin-user-doc -->Return the Feature to be used as the source of the succession of this TransitionUsage,
+     * which is the first ownedMember of the TransitionUsage that is a Feature not owned via a FeatureMembership whose
+     * featureTarget is an ActionUsage. <!-- end-user-doc -->
      *
-     * @generated
+     * @generated NOT
      */
     @Override
     public Feature sourceFeature() {
-        // TODO: implement this method
-        // Ensure that you remove @generated or mark it @generated NOT
-        return null;
+        return this.getOwnedMembership().stream()
```

Comment:
Nope, this is not what is described in the specification.
I will update it to conform to the specification.